addons/addon-base-raas/packages/base-raas-services/lib/environment/environment-mount-service.js

```diff
@@ -330,9 +331,11 @@ class EnvironmentMountService extends Service {
               const statementSidToUse = this._getStatementSidToUse(user.permissionLevel);
               policyDoc.Statement = this._getStatementsAfterAddition(policyDoc, studyPathArn, statementSidToUse);
               policyDoc.Statement = this._ensureListAccess(policyDoc, studyPathArn);
-              if (isStudyAdmin && user.permissionLevel === readWritePermissionLevel) {
-                policyDoc.Statement = this._getStatementsAfterRemoval(policyDoc, studyPathArn, readOnlyStatementId);
-              }
+              policyDoc.Statement = this._getStatementsAfterRemoval(
+                policyDoc,
+                studyPathArn,
+                statementSidToUse === readOnlyStatementId ? readWriteStatementId : readOnlyStatementId,
```

Comment:
Can we add both if conditions- readOnly and readWrite and not assume that !readOnly means readWrite. Just calling this out in case we add some other permissions and we forget to update this.